manim/mobject/svg/text_mobject.py

```diff
@@ -1387,3 +1387,74 @@ def extract_color_tags(self):
 
     def __repr__(self):
         return f"MarkupText({repr(self.original_text)})"
+
+
+@contextmanager
+def register_font(font_file: typing.Union[str, Path]):
+    """Context Manager which adds a font
+    file to Pango's search path.
+    This searches for the font_file at various places.
+
+    The order it searches it described below.
+
+    1. Absolute path.
+    2. In ``assets/fonts`` folder.
+    3. In ``font/`` folder.
+    4. In the same directory.
+
+    .. note ::
+
+        This method of adding font files also works with
+        :class:`CairoText`
+
+    .. important ::
+
+        This method isn't available for macOS. Using this
+        method on macOS will raise an :class:`AttributeError`.
+
+    Parameters
+    ----------
+    font_file :
+        The font file to add.
+
+    Examples
+    --------
+    Use ``with register_font(...)`` to add a font file to search
+    path.
+
+    .. code-block:: python
+       >>> with register_font(...):
+       ...     a=Text("Hello",font=...)
```

Comment:
@behackl do you know how to skip doc test here?

Reply:
Don't write it using the Prompt syntax:

```
.. code-block::

    with register_font(...):
        a = Text("Hello", font=...)
```

I'd also like to see more specific arguments in both cases (e.g., `register_font(path/to/font_file.ttf)`, `font="Custom Font Name"`).

Reply:
done